### PR TITLE
refactor: upgrade jest

### PR DESCRIPTION
**Pull request by anandtiwary**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

**Comment by anandtiwary** [comment]
@aaron-steinfeld  getting this weird error, not sure why

```

 FAIL  projects/hyperdash-angular/src/model/decorators/model-inject.service.test.ts
  ● Test suite failed to run

    ReferenceError: hyperdash_1 is not defined

      30 |
      31 |     @ModelInject(MODEL_API)
    > 32 |     public api!: ModelApi;
         |                  ^
      33 |
      34 |     @ModelInject(INJECT_FOO)
      35 |     public inject1!: string;
      
     ```

**Comment by anandtiwary** [comment]
> @aaron-steinfeld getting this weird error, not sure why
> 
> ```
> 
>  FAIL  projects/hyperdash-angular/src/model/decorators/model-inject.service.test.ts
>   ● Test suite failed to run
> 
>     ReferenceError: hyperdash_1 is not defined
> 
>       30 |
>       31 |     @ModelInject(MODEL_API)
>     > 32 |     public api!: ModelApi;
>          |                  ^
>       33 |
>       34 |     @ModelInject(INJECT_FOO)
>       35 |     public inject1!: string;
>       
>      ```
> ```

nvm fixed the issue

File: package.json [hunk 1]
```diff
@@ -58,17 +57,20 @@
     "cz-conventional-changelog": "^3.3.0",
     "eslint": "^8.28.0",
     "eslint-config-prettier": "^8.5.0",
-    "eslint-plugin-import": "^8.5.0",
-    "eslint-plugin-jsdoc": "^8.5.0",
-    "eslint-plugin-no-null": "^8.5.0",
-    "eslint-plugin-prefer-arrow": "^8.5.0",
-    "eslint-plugin-unicorn": "^8.5.0",
+    "eslint-plugin-import": "2.27.5",
```
Comment (aaron-steinfeld) [inline package.json:60]: these versions seem to have changed wildly - some up, some down. Intended?

File: src/example-dash/example-dash.component.test.ts [hunk 2]
```diff
@@ -1,86 +1,86 @@
...
-      GraphQlDataSourceModule,
-      HttpClientTestingModule
-    ],
-    providers: [{ provide: LoggerService, useValue: { warn: jest.fn() } }]
-  });
-
   beforeEach(waitForAsync(() => {
-    spectator = createComponent();
-    spectator.inject(DefaultConfigurationService).configure();
+    TestBed.configureTestingModule({
```
Comment (aaron-steinfeld) [inline src/example-dash/example-dash.component.test.ts:16]: Why are we removing spectator here?

File: package.json (hunk 1 above)
Comment (anandtiwary) [reply package.json:60]: eslint migration was using "latest" as the version. I replaced it with the exact version that the package-lock resolved..

File: src/example-dash/example-dash.component.test.ts (hunk 2 above)
Comment (anandtiwary) [reply src/example-dash/example-dash.component.test.ts]: not sure what happened here.

File: package.json (hunk 1 above)
Comment (aaron-steinfeld) [reply package.json:60]: we can probably keep them as `^` versions, the pinning is what the lock is for.

**Pull request merged**